crates/services/executor/src/executor.rs

```diff
@@ -1044,302 +1037,223 @@ where
-                .with_policy(ConflictPolicy::Overwrite);
+            self.compute_inputs(inputs.as_mut_slice(), block_st_transaction)?;
 
-            let mut vm_db = VmStorage::new(
-                &mut sub_block_db_commit,
-                &header.consensus,
-                &header.application,
+            let (input, output) = self.execute_mint_with_vm(
+                header,
                 coinbase_contract_id,
-            );
-
-            fuel_vm::interpreter::contract::balance_increase(
-                &mut vm_db,
-                &mint.input_contract().contract_id,
-                mint.mint_asset_id(),
-                *mint.mint_amount(),
-            )
-            .map_err(|e| format!("{e}"))
-            .map_err(ExecutorError::CoinbaseCannotIncreaseBalance)?;
-            sub_block_db_commit.commit()?;
-
-            self.persist_output_utxos(
-                block_height,
                 execution_data,
-                &coinbase_id,
-                block_st_transaction,
-                inputs.as_slice(),
-                outputs.as_slice(),
-            )?;
-            self.compute_state_of_not_utxo_outputs(
-                outputs.as_mut_slice(),
-                inputs.as_slice(),
-                coinbase_id,
                 block_st_transaction,
+                &coinbase_id,
+                &mut mint,
+                &mut inputs,
             )?;
-            let Input::Contract(input) = core::mem::take(&mut inputs[0]) else {
-                unreachable!()
-            };
-            let Output::Contract(output) = outputs[0] else {
-                unreachable!()
-            };
 
-            if execution_kind == ExecutionKind::Validation {
-                if mint.input_contract() != &input || mint.output_contract() != &output {
-                    return Err(ExecutorError::MintMismatch)
-                }
-            } else {
-                *mint.input_contract_mut() = input;
-                *mint.output_contract_mut() = output;
-            }
+            *mint.input_contract_mut() = input;
+            *mint.output_contract_mut() = output;
         }
 
-        let tx = mint.into();
-
-        execution_data.tx_status.push(TransactionExecutionStatus {
-            id: coinbase_id,
-            result: TransactionExecutionResult::Success {
-                result: None,
-                receipts: vec![],
-                total_gas: 0,
-                total_fee: 0,
-            },
-        });
-
-        if block_st_transaction
-            .storage::<ProcessedTransactions>()
-            .insert(&coinbase_id, &())?
-            .is_some()
-        {
-            return Err(ExecutorError::TransactionIdCollision(coinbase_id))
-        }
-        Ok(tx)
+        Self::store_mint_tx(mint, execution_data, coinbase_id, block_st_transaction)
     }
 
     #[allow(clippy::too_many_arguments)]
-    fn execute_chargeable_transaction<Tx, T>(
+    fn execute_mint_with_vm<T>(
```

Comment:
It seems this new function shifted `execute_chargeable_transaction` below and the change diff looks bad now. Could you swap `execute_mint_with_vm` with `execute_chargeable_transaction` please for the history sake